runtime/io.c

```diff
@@ -281,22 +339,33 @@ CAMLexport file_offset caml_pos_out(struct channel *channel)
 
 int caml_do_read(int fd, char *p, unsigned int n)
 {
-  return caml_read_fd(fd, 0, p, n);
+  int r;
+  do {
+    r = caml_read_fd(fd, 0, p, n);
+  } while (r == Io_interrupted);
+  return r;
 }
 
 CAMLexport unsigned char caml_refill(struct channel *channel)
 {
   int n;
-
+ again:
+  check_pending(channel);
   n = caml_read_fd(channel->fd, channel->flags,
                    channel->buff, channel->end - channel->buff);
-  if (n == 0) caml_raise_end_of_file();
+  if (n == Io_interrupted) goto again;
+  else if (n == 0) caml_raise_end_of_file();
   channel->offset += n;
   channel->max = channel->buff + n;
   channel->curr = channel->buff + 1;
   return (unsigned char)(channel->buff[0]);
 }
 
+#define Getch(channel)                                                      \
```

Comment:
Isn't this functionally equivalent to the currently defined `caml_getch` in `io.h`?

Reply:
Yes. It was the same on trunk (https://github.com/ocaml/ocaml/blob/trunk/runtime/io.c#L309) and will get fixed on the rebase: https://github.com/ocaml/ocaml/commit/f7e32a3b4852d65772dc75af72956ee379a2784e#diff-cc528a0b067be28c2081c889ec38bcc70ccc0c42d2e70d3de7b2b4967c0ddd10L82